Remotion/Data/DomainObjects.UnitTests/TestDomainDatabase.cs

```diff
@@ -0,0 +1,103 @@
+
+      lock (s_lock)
+      {
+        if (s_initialized)
+          return;
+
+        s_initialized = true;
+
+        try
+        {
+          StandardConfiguration.EnsureInitialized();
+          TableInheritanceConfiguration.EnsureInitialized();
+
+          SqlConnection.ClearAllPools();
+
+          var masterAgent = new DatabaseAgent(DatabaseTest.MasterConnectionString);
+          masterAgent.ExecuteBatchFile("Database\\DataDomainObjects_CreateDB.sql", false, DatabaseConfiguration.GetReplacementDictionary());
+          var testDomainAgent = new DatabaseAgent(DatabaseTest.TestDomainConnectionString);
+          testDomainAgent.ExecuteBatchFile("Database\\DataDomainObjects_TestDomain_TearDownDB.sql", true, DatabaseConfiguration.GetReplacementDictionary());
+          testDomainAgent.ExecuteBatchFile("Database\\DataDomainObjects_TableInheritanceTestDomain_TearDownDB.sql", true, DatabaseConfiguration.GetReplacementDictionary());
+          testDomainAgent.ExecuteBatchFile("Database\\DataDomainObjects_TestDomain_SetupDB.sql", true, DatabaseConfiguration.GetReplacementDictionary());
+          testDomainAgent.ExecuteBatchFile("Database\\DataDomainObjects_TableInheritanceTestDomain_SetupDB.sql", true, DatabaseConfiguration.GetReplacementDictionary());
+          testDomainAgent.ExecuteBatchFile("Database\\DataDomainObjects_Patches.sql", true, DatabaseConfiguration.GetReplacementDictionary());
+
+          s_standardMappingDatabaseAgent = new StandardMappingDatabaseAgent(DatabaseTest.TestDomainConnectionString);
+          string sqlFileName = StandardMappingTest.CreateTestDataFileName;
+          s_standardMappingDatabaseAgent.ExecuteBatchFile(sqlFileName, true, DatabaseConfiguration.GetReplacementDictionary());
+          string sqlFileName1 = TableInheritanceMappingTest.CreateTestDataFileName;
+          s_standardMappingDatabaseAgent.ExecuteBatchFile(sqlFileName1, true, DatabaseConfiguration.GetReplacementDictionary());
+
+          // We don't want the tests to initialize a default mapping; therefore, modify MappingConfiguration.s_fields.Current so that it will 
+          // throw when asked to generate a new MappingConfiguration.
+
+          var throwingMappingConfigurationContainer = new DoubleCheckedLockingContainer<IMappingConfiguration>(
+              () =>
+              {
+                throw new InvalidOperationException(
+                    "This test failed to setup the mapping configuration. Did you forget to derive from StandardMappingTest or to call base.SetUp?");
+              });
+          var fields = PrivateInvoke.GetNonPublicStaticField(typeof(MappingConfiguration), "s_fields");
+          Assertion.IsNotNull(fields);
+          PrivateInvoke.SetPublicField(fields, "Current", throwingMappingConfigurationContainer);
+        }
+        catch (Exception ex)
+        {
+          Console.WriteLine("TestDomainDatabase.InitializeDatabase failed: " + ex);
+          Console.WriteLine();
+          throw;
+        }
```

Comment:
Some mis-communication:
this try-catch block is needed in SetupFixture, because Nunit swallows meaningful error information that happens during setup fixtures. If we do this during regular testfixture setups, we should get the full details. please verify. It may be that we do not need it at all, now that we do testfixture.setupfxture instead of setupfixture.setup.

Reply:
It is no longer needed in the `SetupFixture`. In this case it probably makes it easier to identify that the problem was a database one if we keep this message here.